docusaurus/docs/tutorials/build-a-streaming-data-source-plugin.md

```diff
@@ -15,143 +14,286 @@ keywords:
+          data: {
+            ...query,
+          },
+        },
+      });
+    });
 
-   return merge(...observables);
-   ```
+    return merge(...observables);
+  }
+```
 
-1. In the `subscribe` function, create a `CircularDataFrame` instance.
+The call to `getGrafanaLiveSrv()` returns a reference to a Grafana backend, and the `getDataStream` call creates the stream. As shown in this code snippet, we need to provide some data to create the stream:
 
-   ```ts
-   import { CircularDataFrame } from '@grafana/data';
-   ```
+- **`scope`** - defines how the channel is used and controlled (specify `data source`)
+- **`namespace`** - unique identifier for our specific data source
+- **`path`** - part that can distinguish between channels created by the same data source. In this example, we want to create a different channel for every query, so we will use the query parameters as part of the path.
+- **`data`** - used to exchange information between the frontend and the backend
 
-   ```ts
-   const frame = new CircularDataFrame({
-     append: 'tail',
-     capacity: 1000,
-   });
+It's worth mentioning that the `path` can be used to exchange data with the backend part as well, but for complex formats, use the `data` property.
 
-   frame.refId = query.refId;
-   frame.addField({ name: 'time', type: FieldType.time });
-   frame.addField({ name: 'value', type: FieldType.number });
-   ```
+:::note
 
-   Circular data frames have a limited capacity. When a circular data frame reaches its capacity, the oldest data point is removed.
+The stream is uniquely identified by a combination of parameters and it will have the format: `${scope}/${namespace}/${path}`. This is only an implementation detail and you probably won't need to know this except for debugging.
 
-1. Use `subscriber.next()` to send the updated data frame whenever you receive new updates.
+:::
 
-   ```ts
-   import { LoadingState } from '@grafana/data';
-   ```
+## Step 3: Set up the plugin backend
 
-   ```ts
-   const intervalId = setInterval(() => {
-     frame.add({ time: Date.now(), value: Math.random() });
+Now we need to add the necessary code to the backend. For that, we change `pkg/plugin/datasource.go`, this is where the backend part responsible for handling the queries created by the frontend is defined. In our case, as we want to handle a stream, we need to implement the `backend.StreamHandler`.
 
-     subscriber.next({
-       data: [frame],
-       key: query.refId,
-       state: LoadingState.Streaming,
-     });
-   }, 500);
+1. Add the following part in the replacement of the `var`:
 
-   return () => {
-     clearInterval(intervalId);
-   };
-   ```
+```
+var (
+    _ backend.CheckHealthHandler    = (*Datasource)(nil)
+    _ instancemgmt.InstanceDisposer = (*Datasource)(nil)
+    _ backend.StreamHandler         = (*Datasource)(nil)
+)
+```
```

Comment:
we could also use syntax highlight in the backend part:
```go
var (
    _ backend.CheckHealthHandler    = (*Datasource)(nil)
    _ instancemgmt.InstanceDisposer = (*Datasource)(nil)
    _ backend.StreamHandler         = (*Datasource)(nil)
)
```

the same for all the others piece of code for the backend

Reply:
Interesting. With regard to syntax highlight, I have two concerns. First, do you feel that the tutorial's text is referring to specific parts of the code block which need to be called out? Second, I want to ensure consistency with other Grafana documentation, so I am researching this further.

Reply:
done.